plugin-utilities-plugin/src/main/kotlin/org/jetbrains/research/pluginUtilities/Preprocessing.kt

```diff
@@ -0,0 +1,56 @@
+package org.jetbrains.research.pluginUtilities
+
+import java.io.File
+import java.util.logging.Logger
+
+/**
+ * Runs multiple preprosessings
+ */
+class Preprocessor(private val preprocessings: List<Preprocessing>) {
+    private val LOG = Logger.getLogger(javaClass.name)
+
+    /**
+     * Preprocesses repository in [repoDirectory] by copying it to [outputDirectory].
+     * Does not mutate the original repository
+     */
+    fun preprocess(repoDirectory: File, outputDirectory: File) {
+        repoDirectory.copyRecursively(outputDirectory)
+        for (preprocessing in preprocessings) {
+            LOG.info("Running preprosessing ${preprocessing.name} for ${repoDirectory.name}")
+            preprocessing.preprocess(outputDirectory)
+        }
+    }
+}
+
+/**
+ * Object that preprocesses a given repository by mutating its file tree
+ */
+interface Preprocessing {
+    val name: String
```

Comment:
Please, don't use such general names for variables. Which name? Or add description in the comments about arguments